packages/amplify-graphql-auth-transformer/src/vtl-generator/ddb/resolvers/mutation.create.ts

```diff
@@ -85,9 +91,14 @@ const iamExpression = (roles: Array<RoleDefinition>, hasAdminRolesEnabled = fals
         );
       }
     });
-  } else {
+  }
+
+  if (genericIamAccessEnabled) {
```

Comment:
As above, is this the right order of evaluation, or should it happen before checking for Admin; and should we only evaluate either the "legacy" or "new" branch rather than both?